src/plugin.rs

```diff
@@ -95,9 +104,33 @@ macro_rules! appbase_plugin_requires {
                return false;
             }
             self.base.state = PluginState::Stopped;
+            $(appbase_plugin_requires_visit!($deps, shutdown);)*
             log::info!("plugin shutdown: {}", <$name>::typename());
             true
          }
       }
    };
 }
+
+#[macro_export]
+macro_rules! appbase_register_async_single {
+   ($self:ident; $run:block;) => {
+      $self.base.handle = Some(tokio::spawn( async move {
+         $run;
+      }));
+   }
+}
+
+#[macro_export]
+macro_rules! appbase_register_async_loop {
+   ($self:ident; $run:block;) => {
+      $self.base.handle = Some(tokio::spawn( async move {
```

Comment:
For blocking task, `tokio::spawn_blocking` should be used.
```suggestion
      $self.base.handle = Some(tokio::spawn_blocking( async move {
```

Reply:
This macro is not for blocking tasks.

Reply:
`tokio::spawn` creates a task that can be processed asynchronously rather than allocating a thread to your `async move {}` directly. A task should be lightweight and yield when can not be executed immediately. Your code contains a loop block and doesn't release the thread, so it's a blocking task by the definition of `tokio::task`.

Refer to:
https://tokio-rs.github.io/tokio/doc/tokio/task/index.html
> When a task cannot continue executing, it must yield instead, allowing the Tokio runtime to schedule another task.